packages/material-ui/src/Typography/Typography.js

```diff
@@ -219,23 +221,7 @@ Typography.propTypes = {
   /**
    * Applies the theme typography styles.
    */
-  variant: PropTypes.oneOf([
-    'h1',
-    'h2',
-    'h3',
-    'h4',
-    'h5',
-    'h6',
-    'subtitle1',
-    'subtitle2',
-    'body1',
-    'body2',
-    'caption',
-    'button',
-    'overline',
-    'srOnly',
-    'inherit',
-  ]),
+  variant: PropTypes.string,
```

Comment:
To keep enough information in the generated API documentation.

```suggestion
  variant: variant: PropTypes.oneOf(PropTypes.string, PropTypes.oneOf([…])),
```